### ISSUE TITLE
Art or illustration for Start/End of game

### ISSUE THREAD

**Issue by ultidonki** [Tuxemon/Tuxemon#2522]
It might be nice to have a unique illustration, or splash screen, for the beginning and end of a campaign.

Discussion from issue #2492:

@JaskRendix 
> I was thinking it would be awesome to create an illustration that wraps up the Spyder campaign, but I don't just want to stop at the ending. I think it would be really cool to create a piece that we could use at the start of the campaign too, maybe even as a background image. That way, we could replace the current 'new game' choice with something more visually appealing. Of course, that would mean we'd need to create a similar artwork for the Xero campaign as well.

@Sanglorian 
> An opening image to advertise the Spyder campaign, and another when the "main" campaign is done would be lovely 

I'm not actually the best at art/illustration/pixel art, so I'm not actually the best person for the job! Other contributors are welcome! :smiley:
(However, I'll make some art if nobody else does :smile: )

Before anyone starts working on it, does anyone have any ideas they want?
Some questions about the art:
- What general idea should the art be of? A landscape, a portrait of a
  character, human characters, tuxemon, tuxemon + characters?
- Does anyone have specific requests - particular characters or tuxemon they
  want to appear (Rockitten, Drokoro, - Billie/Omnichannel CEO)?
- Does the Xero campaign need any yet, or should we wait until the story is a bit longer? @Qiangong2 

Some technical questions:
- Should the art be full-screen, or have space at the bottom for a textbox to appear?
- Should the ending art say 'The end' or 'Congratulations' on it? (But this may need to be changed for different languages)

At this point, anybody can suggest anything, even if it requires lots of art (there's no promise anyone will create all the art, though)

Once an artist starts working on it, it's up to them to decide how much/which parts they want to do, of course.

**Comment by JaskRendix** [comment]
> Should the art be full-screen, or have space at the bottom for a textbox to appear?

Not necessarily, but it needs to be considered that if a textbox appears, that part is going to be hidden, but we can display it entirely too as an image.

> Eventually keeping it in English, then the textbox with the message, which will be translated in different languages.

Eventually keeping it in English, then the textbox with the message, which will be translated in different languages.

**Comment by Sanglorian** [comment]
> @Sanglorian
> 
> > An opening image to advertise the Spyder campaign, and another when the "main" campaign is done would be lovely
> 
> I'm not actually the best at art/illustration/pixel art, so I'm not actually the best person for the job! Other contributors are welcome! 😃 (However, I'll make some art if nobody else does 😄 )
> 
> Before anyone starts working on it, does anyone have any ideas they want? Some questions about the art:
> 
>     * What general idea should the art be of? A landscape, a portrait of a
>       character, human characters, tuxemon, tuxemon + characters?
> 
>     * Does anyone have specific requests - particular characters or tuxemon they
>       want to appear (Rockitten, Drokoro, - Billie/Omnichannel CEO)?

I think that could be up to the artist, but for Spyder the opening art could be of the CEO, since he introduces the player to tuxemon but also turns out to be the villain. Maybe with one of the [new releases](https://wiki.tuxemon.org/Category:Starting_Tuxemon), like Grintot or Budaye? 

Then the closing art could be the player leaving Omnichannel HQ with a big smile, and a Rockitten with them?

> 
> Some technical questions:
> 
>     * Should the art be full-screen, or have space at the bottom for a textbox to appear?
> 
>     * Should the ending art say 'The end' or 'Congratulations' on it? (But this may need to be changed for different languages)
> 

I think the opening art should maybe be small, since I'm imagining it appearing as a pop-up as part of a selection menu - so you scroll between Spyder, Xero, etc, and see the associated image for each pop up as you scroll along. 

I think just having the ending art there will communicate the message. We could have text appear after a moment, saying "Congratulations! You've defeated the bad guys. /n But there's still more to see in the Fondent region." or something like that.

**Comment by ultidonki** [comment]
It's a bit late, and doesn't actually match your request very well, but I drew a close-up of the CEO - is this the kind of thing you wanted, @Sanglorian ? (It doesn't match the game size so I'll need to re-draw it again anyway, think of this as a rough concept image) 

![ceo1](https://github.com/user-attachments/assets/2fa590c4-7e84-4f11-9ec7-2e963fa7adc7)

_Overland sprites by Catch Challenger (Art director Jordan Brule. Project developer Herman Brule.) CC-BY-4.0
Front sprite by Sanglorian under CC-BY-SA-4.0+
This version, by ultidonki, based on the above creators, released under CC-BY-SA-4.0+_

(Feel free to tell me anything about it - Does he look too angry? Not angry enough? Evil? Not evil enough? etc.)

I can also draw a Grintot or Budaye next to him for the final image if needed.

I was imagining this would be used when first starting the game, as the CEO might introduce something. However, now I realize you were both talking about an image to use on the 'new game'/mod select screen, so I don't think this would suit. But you might find a use for an image similar to this so I might finish this. 

> Then the closing art could be the player leaving Omnichannel HQ with a big smile, and a Rockitten with them?

I think this is a good idea, and I want to draw it! Although, there are now 6 different sprites for the player so it might be a bit weird. But, I might do this with the 'white male' sprite. If I add the player in on a different layer, it would be easy to change it to another sprite later, even by someone who's not good at art. So I'm thinking of doing that. (Same with the starter Tuxemon, Rockitten, I guess - I could put that on another layer so whichever one the player chose at the beginning, appears - unless it evolves, as I'd need to draw all evolutions just in case... Let me think about it a bit more!)

**Comment by Sanglorian** [comment]
Very expressive, thank you! I think it works well, although the tie doesn't read as a tie to me - maybe if it flared out instead of curving back in? 

That's a good point about not knowing who the hero will be or which monster they'll choose. I think white male + Rockitten works well for starters, and as you say others could add other options as we go along.

**Comment by ultidonki** [comment]
Thank you for the reply :smile: 

I'm not sure when I'll work on this, but one day I'll just feel in the mood and sit down and do it. (I haven't drawn any Tuxemon art recently but I might start doing it again in the next month or two) (I also haven't done any tuxemon illustrations for seasons eg. Christmas or number of stars, as I see it's almost reached 1000 on Github here!)

I already have a very-rough drawing of the omnichannel HQ for the ending, so I just need to draw the player/billie/shop assistant etc. I'll post it here when it's still rough so people can provide feedback. Like I say it might be a month or two, or more.

**Comment by HippasusTwo** [comment]
I like the idea of seeing the player by HQ for an end scene. What you could do is focus on HQ first, and then simply use a code command to splice in the usual character sprite and the first mon in their party: no need to re-draw every single possible starter mon, and this way people who abandoned their starter in favor of something else will see their favorite mon at the end too. Then when artists have the time, they can do specialized versions of the characters/mons if they feel like it.

I was actually thinking some kind of simple animation would be really cool, like a 'legendary' running or flying, or two characters fighting for the start screen. Start animations don't necessarily have to make much sense or tie into anything, they can just look cool and show off game concepts, but obviously it would be nice to have it tie into something as well.

**Comment by ultidonki** [comment]
![tuxemon_end_concept1](https://github.com/user-attachments/assets/a14fa7ab-2047-41eb-a74a-b82425b5eb8a)
(All licenses for these characters/tuxemon are available on the wiki - https://wiki.tuxemon.org/ - I'll make proper license texts before finishing the work)

Here are some concepts for the ending screen. The actual images will be smaller resolution, of course, but they should look similar. If anyone wants different poses for the characters or Tuxemon, let me know. Otherwise, I'll try turning these into rough images at the correct resolution size, and get more feedback before I make the final versions. 

> What you could do is focus on HQ first, and then simply use a code command to splice in the usual character sprite and the first mon in their party

That's a good idea - in fact, you could do every tuxemon in their party, or something like that. 

> I was actually thinking some kind of simple animation would be really cool, like a 'legendary' running or flying, or two characters fighting for the start screen.

I agree, that would be very cool. But a lot of hard work. :smile: If someone wanted to make an animation, I'm sure it would be approved (within reason). Or if somebody wanted to get the ball rolling by suggesting ideas for an animation (eg. Drokoro flying, Rockitten eating grass, Propellercat flying out of a bush) or something. I'm not sure how much help I'll be though. :smiley:

That reminds me - we still have a grey/black splash screen, that says 'Made with Python' when the game starts up. We talked about adding some art/animation in there at some point, but I never drew anything and I don't think anyone else did either. So if anyone wants to draw something for that, we could put some art/animation there.

**Comment by HippasusTwo** [comment]
Looking nice!

I would be willing to make an animation, as I've animated animals running/flying/eating before and know how that works (and if I forget it's actually pretty easy to find example animation framesets on opengameart): a really simple version could easily be just 4-6 frames on loop, although the better looking ones have more like 12 frames. But I am unsure of exactly what to animate. I'll have to think about it.

I profess I always loved the opening sequences of early game boy or playstation games, which often had dramatic music that seemed to hit important beats at the same time important shifts in the animation occurred. I haven't looked at the music much but there has to be one that has a nice 'intro theme' feel to it, preferably around 30-60 seconds or so. If I'm feeling more ambitious I think an interactive scene between multiple characters would be a really nice touch.

Do Rockittens eat grass? I guess as cats they'd like catnip. Hm. Maybe glowing eyes in a bush, lured by a thrown little box of treats, revealed to be a cute little kitty... only for another character(s) to appear, battle ensues. Or something more mellow, depending on the music.
If we wanted to go really sad we could go with Tuxemon starters (or maybe just one, like Rockitten) looking expectantly out of the window, days passing, only to get thrown out into the trash...

**Comment by Sanglorian** [comment]
I can imagine Rockitten eating kibble or meat. I've also imagined them eating stones, including semi-precious and precious stones.

**Comment by ultidonki** [comment]
I've made a rough version of the drawings I made earlier: (Krita has a pixel art tool, so I want to draw them again using that just to make them a bit better)

![Image](https://github.com/user-attachments/assets/573ceb38-cc2f-4d25-8d0d-1c106f1adfed)

And here's a rough animation I've made, which I could clean up and maybe could be used if people like it? 

(It looks a bit blurry when I exported it, but it's done at the correct pixel size:)

https://github.com/user-attachments/assets/7b5d7055-625d-4819-b0c8-a62c46313129

And here's a version without any trees or anything, so you can see the character better:

https://github.com/user-attachments/assets/100139aa-ae39-4261-95c2-a73eb709632e

**Comment by JaskRendix** [comment]
@ultidonki as you can see, we have the opportunity to integrate it after the splash screen, which features Tuxemon and the Pygame logo, and before the start screen

https://github.com/user-attachments/assets/f6526f44-8435-4cde-9f5d-e21db3472e2f

**Comment by ultidonki** [comment]
Nice! It's looking good! :smiley: 

Does pygame/Tuxemon play .mp4's? I didn't know that (I thought it all had to be animated .png's). 
However, I can export the animation as .png's when I'm finished, if that's easier. 

I was thinking of adding a Tuxemon flying or running beside the man, as if he was chasing it - but maybe I can add a Rockitten running too or something? 😄

**Comment by JaskRendix** [comment]
@ultidonki my apologies for the confusion. Pygame itself doesn't directly play .mp4 files. I was referring to the Pygame logo that appears on black screen before this. What I actually did was break down the .mp4 into individual frames, resulting in those 14 PNG files you saw. Then, I used the animation capabilities built into the Tuxemon game engine to string those PNG frames together and create the animation.

Just a thought, that cloud going over the skyscraper kind of stands out. Maybe having it pass behind would look a little more natural?

You know what might look cool? A Rockitten running on the left, and then maybe a Paragrin flying around above the trainer.

**Comment by ultidonki** [comment]
> Pygame itself doesn't directly play .mp4 files.

No problem, that's what I was expecting! 

> Just a thought, that cloud going over the skyscraper kind of stands out. Maybe having it pass behind would look a little more natural?

Yes, I think I'll try to put it behind when I make a better version :smile: 

> You know what might look cool? A Rockitten running on the left, and then maybe a Paragrin flying around above the trainer.

Good idea 😃 I'll add it to my list of ideas I might do at some point!

**Comment by ultidonki** [comment]
New version of the animation with a running Rockitten: (I've started adding a bird too, but that'll take a lot longer I think!)

https://github.com/user-attachments/assets/9895cafa-ca52-4799-bda9-640db91e14b1

**Comment by ultidonki** [comment]
It's not exactly a Pairagrin (probably more like a Cardiling) but I added a bird flying above the trainer now too. 

I still have a long way to go before it's ready to go in the game, though. :smiley:

https://github.com/user-attachments/assets/5870641a-cded-4fed-acba-9509529c71de

**Comment by HippasusTwo** [comment]
It's nice! I like the way it swoops.

**Comment by ultidonki** [comment]
Here's a rough version of the 3 heroes at the end: (Not completed yet, I want to make the lines neater a bit first)

<img width="240" height="160" alt="Image" src="https://github.com/user-attachments/assets/972d7b7a-61cb-4915-bd7f-e4aa790dd868" />

**Comment by JaskRendix** [comment]
@ultidonki I like it!

**Comment by ultidonki** [comment]
I've cleaned up the 3 main character's sprites (a tiny bit!), I think this is 'good enough' for me - (of course if anyone else wants to change anything or try it themselves, feel free!) 

Also feel free to change the file names, or crop the images if that's easier (ie. so each character doesn't have any space around the outside) (Also, I think these images are currently 240x160 which I think is the wrong resolution anyway)

Next thing for me to do is either the Tuxemon, or the background of Omnichannel HQ!

<img width="240" height="160" alt="Image" src="https://github.com/user-attachments/assets/3462d900-c1ee-45a1-afba-7897f18116a5" />

<img width="240" height="160" alt="Image" src="https://github.com/user-attachments/assets/4ec48dfa-71d4-4b66-993a-1ebe67d58408" />

<img width="240" height="160" alt="Image" src="https://github.com/user-attachments/assets/bb84ca24-6c6b-432b-9c65-c13f4e21b41d" />

**Comment by Sanglorian** [comment]
Really nice, thank you! I've added them to the wiki to illustrate the different trainer classes.

**Comment by JaskRendix** [comment]
@ultidonki 
<img width="240" height="160" alt="Image" src="https://github.com/user-attachments/assets/0211ed6c-9ef8-4aaf-bbe3-98564fae2873" />

**Comment by ultidonki** [comment]
> [@ultidonki](https://github.com/ultidonki) <img alt="Image" width="240" height="160" src="https://private-user-images.githubusercontent.com/64643719/491149572-0211ed6c-9ef8-4aaf-bbe3-98564fae2873.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk2MTAzNzQsIm5iZiI6MTc1OTYxMDA3NCwicGF0aCI6Ii82NDY0MzcxOS80OTExNDk1NzItMDIxMWVkNmMtOWVmOC00YWFmLWJiZTMtOTg1NjRmYWUyODczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA0VDIwMzQzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlMzQxNjBlNzE0Njc2ZDJlMGRhNWQ1NTI2NDM4NmZmN2NlMmRjNmE4YTYyZGMyZDhlMjFiNDVkZmQ4MTc4NjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.K1B26F2mivxXeB6yKU0b1fCdT8BgkdLzejhD3AY0I-o">

Thank you, it looks cute. Did you use AI or something to generate it?

**Comment by ultidonki** [comment]
I've finally tried to draw the Omnichannel skyscraper background (for the very end of the game): The idea is that at first the game will show the top of the skyscraper, and then scroll down to see the characters at the bottom. Although I'm not sure how it will look in the game... 

Anyway, I did 2 versions - one normal, and one where the skyscraper is darker at the top, to make it look more ominous. If people prefer one to the other, or if someone wants to make a better version, feel free. 😃 

<img width="240" height="640" alt="Image" src="https://github.com/user-attachments/assets/c8fc6c17-3f5b-44d8-b0d0-dc221916b894" />

<img width="240" height="640" alt="Image" src="https://github.com/user-attachments/assets/7f6c22c3-6f5d-4652-83a4-416ba1754676" />
<img width="240" height="640" alt="Image" src="https://github.com/user-attachments/assets/e0d74ac2-ffea-41d5-a05d-610f3be2b768" />
<img width="240" height="640" alt="Image" src="https://github.com/user-attachments/assets/8d1772ce-1972-4f06-ba0b-ebc55cf6a611" />

**Comment by Sanglorian** [comment]
I love it! I think the bright one works for the victorious end of the game.

**Comment by ultidonki** [comment]
https://github.com/user-attachments/assets/a2aae615-8247-4e50-9614-16298098aad2

This animation is almost finished. :smiley:  Ignore the right half, I couldn't figure out how to remove that when exporting 😊 

@JaskRendix 
What sort of format do you want this animation in, if it's to be played in the game? (We might not have the ability to do animations like this yet, maybe?)
Just an .mp4 or a .gif file, or a bunch of individual .png's, eg. trainer_run_01.png, trainer_run_02.png... ? 
The clouds and stuff are just being panned a few pixels to the right each time. 

I will need to do a bit more work to finish it properly, but if you let me know how you want the finished format, I can export it when it's complete. 🥳

**Comment by JaskRendix** [comment]
@ultidonki That looks great! Don't worry about the right half for now.

Right now, our animations are based on image sheets (like mods/tuxemon/animations/intro/intro.png).
A folder of individual .png files (e.g., trainer_run_01.png, trainer_run_02.png) would be ideal. We can just run a script to stitch them into the final sheet!

**Comment by ultidonki** [comment]
Understood! I see how that works. :smile:
Oh, there's already an intro, though (I haven't played since it was added!)
But, I'll finish this at some point and make a pull request when done. 🫡

**Comment by Sanglorian** [comment]
> Understood! I see how that works. 😄 Oh, there's already an intro, though (I haven't played since it was added!) But, I'll finish this at some point and make a pull request when done. 🫡

Thanks heaps, this is great! We do have an intro, but this could work great as an end credits for one or more of the campaigns?

**Comment by HippasusTwo** [comment]
Yes, we need an end credits animation, I think yours would fit as a nice 'default' that could be over-ridden for a given campaign.

What I would do is combine the 'scroll/pan down the omnicorp building' animation and the running animation, since end credits are rarely just a single animation, esp. when you have a LOT of credits. And I might also have a few pan ins of the sprites of whatever monsters you chose on your team, or random monsters if you only ever had one monster.

edit: the final anim may be a bunch of pngs, but for modifying it would be best to have an additional file with the different layers even if that file is technically never used in-game. This would allow someone to potentially go in and later create, say, the character running against a different background, such as a desert or snowy mountain. Usually the running / flying is the hardest part of the animation and backgrounds are easy in comparison.
